### PR TITLE
MEED-291 Add Market Capitalization public endpoint 

### PR DESCRIPTION
**Pull request by ArfaouiMohamedAmine**
This change will introduce a new REST endpoint to retrieve market capitalization of Meeds Token. This endpoint can be used and consulted from other third party exchange sites.

### CODE REVIEW

File: deeds-dapp-service/src/main/java/io/meeds/deeds/service/ExchangeService.java [hunk 1]
```diff
@@ -47,52 +47,52 @@
 @Component
 public class ExchangeService {
 
-  private static final Logger            LOG                               = LoggerFactory.getLogger(ExchangeService.class);
```
Comment (boubaker) [inline deeds-dapp-service/src/main/java/io/meeds/deeds/service/ExchangeService.java]: please apply formatting settings in your IDE https://developer.exoplatform.org/#id-ide-settings-to-work-on-exo-platform-projects and avoid formatting all class, only added code

File: deeds-dapp-service/src/main/java/io/meeds/deeds/service/MeedTokenMetricService.java [hunk 2]
```diff
@@ -56,6 +66,9 @@ public class MeedTokenMetricService {
   @Getter
   private List<String>               lockedPolygonAddresses;
 
+  @Value("${meeds.exchange.Coingecko.UsdPrice:https://api.coingecko.com/api/v3/simple/price?ids=meeds-dao&vs_currencies=USD}")
+  private String InstantMeedUsdPrice;
```
Comment (boubaker) [inline deeds-dapp-service/src/main/java/io/meeds/deeds/service/MeedTokenMetricService.java]: Formatting issue and the variable name should start with a lower case, in addition, the name isn't significant IMO 
```suggestion
  private String meedUsdPriceProviderUrl;
```

File: deeds-dapp-service/src/main/java/io/meeds/deeds/service/MeedTokenMetricService.java [hunk 3]
```diff
@@ -101,6 +117,9 @@ public void computeTokenMetrics() {
     Map<String, BigDecimal> lockedBalances = getLockedBalances();
     metric.setLockedBalances(lockedBalances);
 
+    BigDecimal marketCap = getMarketCapitalization();
+    metric.setMarketCapitalization(marketCap);
```
Comment (boubaker) [inline deeds-dapp-service/src/main/java/io/meeds/deeds/service/MeedTokenMetricService.java]: you missed to add the `meedUsdPrice` in `metric`

File: deeds-dapp-service/src/main/java/io/meeds/deeds/service/MeedTokenMetricService.java [hunk 4]
```diff
@@ -156,6 +175,42 @@ public Map<String, BigDecimal> getLockedBalances() {
     return lockedBalances;
   }
 
+  /**
+   * @return {@link BigDecimal} . This will return the Makert Capitalization value of the Meeds Token
+   * throughout the formula of MarketCap = TotalSupply * Meeds price
+   */
+  public BigDecimal getMarketCapitalization() {
```
Comment (boubaker) [inline deeds-dapp-service/src/main/java/io/meeds/deeds/service/MeedTokenMetricService.java]: You should add this operation in `ExchangeService` to have a single reference where to get the Meed USD price

File: deeds-dapp-service/src/test/java/io/meeds/deeds/service/MeedTokenMetricTest.java [hunk 5]
```diff
@@ -44,12 +44,12 @@
     }
 )
 @TestPropertySource(
-    properties = {
```
Comment (boubaker) [inline deeds-dapp-service/src/test/java/io/meeds/deeds/service/MeedTokenMetricTest.java]: formatting issues

File: deeds-dapp-service/src/main/java/io/meeds/deeds/service/ExchangeService.java (hunk 1 above)
Comment (boubaker) [reply deeds-dapp-service/src/main/java/io/meeds/deeds/service/ExchangeService.java]: > and avoid formatting all class, only added code

File: deeds-dapp-service/src/main/java/io/meeds/deeds/service/MeedTokenMetricService.java (hunk 4 above)
Comment (ArfaouiMohamedAmine) [reply deeds-dapp-service/src/main/java/io/meeds/deeds/service/MeedTokenMetricService.java]: I made two seperate function from getMarketCapitalization() -> getMarketCapitalization + getMeedUsdPrice() inorder to inject the price in the metric seperately ! 
Still need the change you proposed ?

**Review by boubaker — APPROVED** [review]

**Pull request merged**